src/Database/DatabaseConfig.php

```diff
@@ -114,6 +120,43 @@ public function connection(): array
         );
     }
 
+    /**
+     * Tenant's host database connection config.
+     */
+    public function hostConnection(): array
+    {
+        $config = $this->tenantConfig();
+        $template = $this->getTemplateConnectionName();
+        $templateConnection = config("database.connections.{$template}");
+
+        if ($this->manager() instanceof Contracts\ManagesDatabaseUsers) {
+            // We don't need username and password for database creation/deletion
+            // Username and password will be saved in Tenant's config
+            // and will be used for tenancy db connection
+            unset($config['username']);
+            unset($config['password']);
+        }
+
+        if (! $config) {
+            return $templateConnection;
+        }
+
+        return array_replace($templateConnection, $config);
+    }
+
+    /**
+     * Purge host database connection.
+     */
+    public function purgeHostConnection(): void
+    {
+        $tenantHostConnectionName = $this->getTenantHostConnectionName();
+
+        if (config("database.connections.{$tenantHostConnectionName}")) {
+            DB::purge($tenantHostConnectionName);
+            config(["database.connections.{$tenantHostConnectionName}" => null]);
+        }
```

Comment:
I think there should be a better check here. Just because the config value exists doesn't mean that Laravel has the PDO connection cached. See at the method which purges the `tenant` connection.

Also it's possible that we can completely remove the check and just always reset the config and the connection cache.

And I believe we have `DatabaseManager` injected so we shouldn't be using the `DB` facade here.

Reply:
> I think there should be a better check here.

I see. So I checked the `purge` method. It is disconnecting the `PDO` and then removing the connection of The active connection instances array. `purge` method looks like this. 
```php
public function purge($name = null)
    {
        $name = $name ?: $this->getDefaultConnection();

        $this->disconnect($name);

        unset($this->connections[$name]);
    }
```

 
So we can modify our logic. 
```php
if (array_key_exists($tenantHostConnectionName, $databaseManager->getConnections())) {
        $databaseManager->purge($tenantHostConnectionName);
    }
```
which is working fine. 

> Also it's possible that we can completely remove the check and just always reset the config and the connection cache.

And I tried this. If we completely remove the connection, tests are still passing. But my concern is `purge` method check if the given connection name is null, it purges the default connection. And the host connection can be null, mostly when hitting the method the first time in the request life cycle. 

So I think the `if` check is better since we want to purge only the host connection. 


> And I believe we have DatabaseManager injected, so we shouldn't be using the DB facade here.

No, we don't have. We can inject this or keep using the facade.

Reply:
> And the host connection can be null, mostly when hitting the method the first time in the request life cycle.

Can you expand on this? I think it shouldn't be null since the method we have for returning the name of the connection has a `: string` typehint

> No, we don't have. We can inject this or keep using the facade.

Ah I see, this is the `DatabaseConfig`. Makes sense, facade is fine then.

Reply:
> Can you expand on this? I think it shouldn't be null since the method we have for returning the name of the connection has a : string typehint

I mean getting the connection from the config `config("database.connections.{$tenantHostConnectionName}")` can be null at first time. 
For example if we perform this `$databaseManager->purge(config("database.connections.{$tenantHostConnectionName}"));` without `if` check, config will return the `null` and `purge` will purge the default connection since it received `null`.

Reply:
What do you mean specifically by _"mostly when hitting the method the first time in the request life cycle"_?

And I think the host connection should **only** be used for logic related to creating databases? So I'm not sure why it'd be null or not null at different points in the request lifecycle. I don't think it would be even used.

Reply:
> What do you mean specifically by "mostly when hitting the method the first time in the request life cycle"?

As you said "host connection should only be used for logic related to creating databases" that's correct. But CreateDatabase fetches hostManager twice. One for ensuring a database can be created and one for actually creating a database. So when it fetches the host manager first time, it tries to purge the connection but the connection is not created yet. 

But I see that Laravel's underline method is doing the same. eg purging connection key only it is exists in connections array.  
```php
if (isset($this->connections[$name = $name ?: $this->getDefaultConnection()])) {
    $this->connections[$name]->disconnect();
}
```
So checking on our side doesn't make sense, and I'm removing it.

Reply:
So this is resolved now?

Reply:
> So everything in this thread is resolved now?

Yes.